src/main/java/kostka/moviecatalog/controller/UserController.java

```diff
@@ -58,4 +65,16 @@ public String getAllMoviesPage(final Model model) {
         model.addAttribute(ALL_MOVIES_KEY, cacheService.getMoviesFromCacheWithKey(ALL_MOVIES_KEY));
         return "allMovies";
     }
+
+    @GetMapping("randomMovie")
+    public String getRandomMovieDetail(final Model model) {
+        LOGGER.info("getting random movie detail page");
+        try {
+           Long randomMovieId = randomMovieService.getRandomMovieIdFromAllMovies();
+           return REDIRECT_MOVIE_DETAIL_VIEW + randomMovieId;
+        } catch (NoMoviesInDbException e) {
```

Comment:
OPTIONAL: Maybe you can introduce an exception handler to redirect to all movies page in all cases where the exception is thrown.

Reply:
Will do it in separate story :-)